benchmarks/src/main/scala-2.12/zio/GenBenchmarks.scala

```diff
@@ -0,0 +1,92 @@
+package zio
+
+import java.util.concurrent.TimeUnit
+
+import org.openjdk.jmh.annotations._
+import org.scalacheck
+import zio.test.Gen
+import zio.IOBenchmarks.unsafeRun
+
+@State(Scope.Thread)
+@BenchmarkMode(Array(Mode.AverageTime))
+@OutputTimeUnit(TimeUnit.NANOSECONDS)
+class GenBenchmarks {
+
+  @Param(Array("1000"))
+  var listSize: Int = _
+
+  @Param(Array("1000"))
+  var elementSize: Int = _
+  @Benchmark
+  def zioDouble: List[Double] =
+    unsafeRun(Gen.listOfN(listSize)(Gen.uniform).sample.map(_.value).runHead.get)
```

Comment:
I don't know if it will make much of a difference but instead of doing `Gen.listOfN` and then `runHead I think I would do `Gen.uniform.sample.map(_.value).take(1000).run`. In other words, generate doubles and generate 1000 of them instead of Generating just one big list. You can do the same thing with the other benchmarks.